pkg/controller/kogitoapp/resource/create_resources.go

```diff
@@ -140,6 +159,62 @@ func imageStreamBuilder(chain *builderChain) *builderChain {
 	return callPostCreate(created, is, chain)
 }
 
+// Assure that all Kogito images are available before the build.
+func kogitoImageStreamBuilder(chain *builderChain) *builderChain {
+	created := true
+	var need2create []string
+
+	ocp := openshift.ImageStreamC(chain.Context.Client)
+
+	for _, key := range ImageStreamNameList {
+		isTag, _ := ocp.FetchTag(types.NamespacedName{Name: key, Namespace: ImageStreamNamespace}, ImageStreamTag)
+		if isTag == nil {
+			need2create = append(need2create, key)
+		}
+	}
+
+	if len(need2create) > 0 {
+		imageStreamLocation := fmt.Sprintf(ImageStreamLocation, ImageStreamTag+"/1")
+		log.Infof("Using imagestream %s", imageStreamLocation)
+		resp, err := http.Get(imageStreamLocation)
+		if err != nil {
+			log.Errorf("Failed to fetch imagestream %s", err.Error())
```

Comment:
In case error happens it is logged, however the method processing continue. Is it expected behaviour?

Reply:
not really.
for now it is returning the error for the reconciler, it will fail and no build will start.